releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
+and backported to `release-x.y`. For this reason, the later we branch, the less extra work
+will be necessary to produce a stable release.  
+
+### Maintaining a stable branch
+
+Any patches that are necessary for fixing bugs in a stable branch will need to be merged
+first into master, and the later stable branches.
+
+`release-0.8`  <-- `release-0.9` <-- `devel`
+
+Patches could be simple cherry-picks, or sometimes they could need specific adaptations
+to the branch.
+
+When a patch is merged to a stable branch the images for such branch are updated, for example
+`quay.io/submariner/submariner:release-0.9`
+
+### How long is a stable branch maintained
+
+Initially stable branches will be maintained until the next stable release is released,
+this would change in the future if we introduce the concept of long term support releases.
+
+### Reducing the cadence of minor releases
+
+Based on the previous ideas the existing minor version cadence doesn't seem realistic
+for maintaining stable branches. We don't want fake stable branches where patches
+are never backported.
+
+With the new cadence after each sprint we would produce a milestone release,
+for example
+
+* `0.9-milestone-1-rc0`, ... , `0.9-milestone-1`
```

Comment:
Maybe it would make things simpler to just call these RCs? That's effectivally what they are, snapshots that folks can use to test/preview the upcoming release. It would also keep our get.submariner.io logic untouched, just use the existing RC flag to get the latest milestone/RC.

Reply:
Not a bad idea but it conflicts with some clatifications  I added above based on comments (like branching off right before `rc0` is created).

We can add logic to grab milestones to get.submariner.io like VERSION=milestone, would that be ok?

Reply:
That's not the purpose of RC though. RC is "release candidate" so in the end you're expected to release a GA.

Probably since it's milestones and not actual releases it'll make more sense to go with `pre0`, `pre1` etc till the actual milestone gets released (as a pre-release itself).
This way we're not implying an imminent release.